frontend/packages/dev-console/src/components/svg/SvgBoxedText.tsx

```diff
@@ -16,95 +12,83 @@ export interface SvgBoxedTextProps {
+  const iconSpace = kind && iconSize ? iconSize.width + paddingX : 0;
 
-  componentDidUpdate({ className, children }: SvgBoxedTextProps) {
-    if (this.props.children !== children || this.props.className !== className) {
-      this.computeBoxSize();
-    }
-  }
-
-  private computeBoxSize() {
-    const { current } = this.textRef;
-    if (current && current.getBBox) {
-      this.setState({ bb: current.getBBox() });
-    }
-  }
-
-  render() {
-    const {
-      children,
-      className,
-      paddingX = 0,
-      paddingY = 0,
-      cornerRadius = 4,
-      x = 0,
-      y = 0,
-      kind,
-      onMouseEnter,
-      onMouseLeave,
-      ...other
-    } = this.props;
-    const { bb } = this.state;
-    const iconSpace: number =
-      kind && this.iconRef.current ? this.iconRef.current.getBBox().width + paddingX : 0;
-
-    return (
-      <g className={className} onClick={(e) => e.stopPropagation()}>
-        <SvgDropShadowFilter id={FILTER_ID} />
-        {bb && (
-          <rect
-            filter={createSvgIdUrl(FILTER_ID)}
-            x={x - paddingX - bb.width / 2 - iconSpace / 2}
-            width={bb.width + paddingX * 2 + iconSpace}
-            y={y - paddingY - bb.height / 2}
-            height={bb.height + paddingY * 2}
-            rx={cornerRadius}
-            ry={cornerRadius}
-          />
-        )}
-        {bb && kind && (
-          <SvgResourceIcon
-            ref={this.iconRef}
-            x={x - bb.width / 2 - paddingX / 2}
-            y={y}
-            kind={kind}
-          />
-        )}
-        <text
-          ref={this.textRef}
-          {...other}
-          x={x + iconSpace / 2}
+  return (
+    <g className={className} ref={typeof truncate === 'number' ? labelHoverRef : undefined}>
+      <SvgDropShadowFilter id={FILTER_ID} />
+      {textSize && (
+        <rect
+          filter={createSvgIdUrl(FILTER_ID)}
+          x={x - paddingX - textSize.width / 2 - iconSpace / 2}
+          width={textSize.width + paddingX * 2 + iconSpace}
+          y={y - paddingY - textSize.height / 2}
+          height={textSize.height + paddingY * 2}
+          rx={cornerRadius}
+          ry={cornerRadius}
+        />
+      )}
+      {textSize && kind && (
+        <SvgResourceIcon
+          ref={iconRef}
+          x={x - textSize.width / 2 - paddingX / 2}
           y={y}
-          textAnchor="middle"
-          dy="0.35em"
-          onMouseEnter={onMouseEnter}
-          onMouseLeave={onMouseLeave}
-        >
-          {children}
-        </text>
-      </g>
-    );
-  }
-}
+          kind={kind}
+        />
+      )}
+      <text
+        {...other}
+        ref={textRef}
+        x={x + iconSpace / 2}
+        y={y}
+        textAnchor="middle"
+        dy="0.35em"
+        onMouseEnter={onMouseEnter}
+        onMouseLeave={onMouseLeave}
+      >
+        {typeof truncate === 'number'
+          ? labelHover
+            ? children
+            : truncateEnd(children, truncate)
```

Comment:
New functionality?

Reply:
Moved code from old `WorkloadNode` into here so it's reusable for all node labels.